server/app/services/auth_provider.rb

```diff
@@ -35,48 +30,65 @@ class AuthProvider < OpenStruct
       :root_url
   ]
 
-  def self.reset_instance
-    Singleton.send :__init__, self
-    self
+  attr_accessor :client_id
+  attr_accessor :client_secret
+  attr_accessor :authorize_endpoint
+  attr_accessor :code_requires_basic_auth
+  attr_accessor :token_endpoint
+  attr_accessor :token_method
+  attr_accessor :token_post_content_type
+  attr_accessor :userinfo_scope
+  attr_accessor :userinfo_endpoint
+  attr_accessor :userinfo_username_jsonpath
+  attr_accessor :userinfo_email_jsonpath
+  attr_accessor :userinfo_user_id_jsonpath
+  attr_accessor :root_url
+  attr_accessor :cloud_api_url
+  attr_accessor :ignore_invalid_ssl
+  attr_accessor :provider_is_kontena
+
+  def self.instance
```

Comment:
A static method named `instance` makes me think this is holding on to a lazy-loaded singleton, not creating a new instance each call.